xtask/src/tools/lychee.rs

```diff
@@ -2,14 +2,11 @@ use crate::utils::PKG_PROJECT_ROOT;
 
 use anyhow::{anyhow, Result};
 use camino::Utf8PathBuf;
-use lychee_lib::{
-    Client, ClientBuilder, Collector, FileType, Input, InputSource, Request,
-    Result as LycheeResult, Uri,
-};
-use reqwest::StatusCode;
+use futures::TryStreamExt;
+use http::StatusCode;
```

Comment:
Can we also update the workspace `reqwest` to 0.12 so we don't have two versions of both that and `http` floating around? Or is that a big lift?

Reply:
I don't think it's that much of a lift, but I was hoping to limit this PR to lychee. That way, we have a cleaner paper trail, cognitive overhead and smaller blast radius (since this should be strictly xtask)

We do have reqwest marked [here](https://github.com/apollographql/rover/pull/1755), so I can follow this up immediately with that.